### ISSUE TITLE
Number Translation on Switch - Gateway menu

### ISSUE THREAD

**Issue by HorselessName** [iNextrix/ASTPP#543]
I have two gateways from two different providers, both using Tech Prefix for auth.
If i use the menu "Carriers - Trunk" and set the second one as "Failover GW #2" it'll not work because Number Translation is set for the first try only, and the Tech Prefix for the second try is different from the first one...

The only option i could think of is "Dialplan Variable" but i added 55/TechPrefix55 like i did on Number Translation for the trunk and did'nt work.

Help ?

**Comment by smrdoshi** [comment]
Hi @HorselessName 

In your case, you will require to setup 2 trunks and under each trunk set 1 gateway and in rate group select both trunks. 

For number translation, you can use prepend and strip field in termination rates or use localization concept and select it in provider. 

That will resolve your issue.

**Comment by HorselessName** [comment]
Right. So i have to add the Tech Prefix on each destination on the Termination Rates?
Lets say if i have 10000 prefixes, i need to edit them all and add the tech prefix for each one ?

I'll try that and add a edit saying if it works ...

**Comment by HorselessName** [comment]
Are there any examples on how to use prepend and strip field?

**Comment by HorselessName** [comment]
Are there any examples on how to use prepend and strip field?
For example, if i have the prefix 55419 and want to convert to 353555419 ?

**Comment by HorselessName** [comment]
Right...
I'll test it and edit this comment here saying if it works.

**Comment by HorselessName** [comment]
Hello... @smrdoshi 
Unfortunately the prepend field will not help in this scenario.
For example, i need to call to 5541991919191 and the first try goes to one provider...
The failover goes to another provider...

Even if i insert the tech prefix for the first provider, let's say 91915541991919191, if the call fails and goes to the second provider on the failover, the number that'll go to that provider is 91915541991919191, and that prefix is different from the prefix of that provider.

It would be nice if we could add a prefix directly to the provider... If we have multiple providers with different prefixes, we aren't going to be able to differ which is which...

**Comment by MrSupaMike** [comment]
Hello,

@smrdoshi is right, I've been using on version v3.6 without a problem. Setting up number translation at each trunk and setting multiple trunks in rate groups does the fail-over with correct prefix for each provider. I don't know if it's the same on the latest version.

**Comment by HorselessName** [comment]
@MrSupaMike It is'nt doing the fail-over here...
I configured three trunks with specific number translation on each ( Each trunk is configured to do one try only on the provider gateway tho, since each provider gave me one ip only ), and selected them all in the same rate group, but the call does'nt pass to the next selected trunk when the call fails...

If i remove the number translation and set multiple trunks on one gateway, the failover works. It goes to the second try but the provider receives the wrong tech prefix...

How did you do that? @MrSupaMike Each trunk you have is trying one time only, right?
Like this:
First trunk: https://i.imgur.com/WccVF0z.png
Second trunk: https://i.imgur.com/3kyBAqP.png
Rate Group with both selected: https://i.imgur.com/2iRTL0Q.png

What can i do?

**Comment by HorselessName** [comment]
![image](https://user-images.githubusercontent.com/20173635/65042255-d95d3c80-d92e-11e9-80b7-ce94769a1f9e.png)

**Comment by HorselessName** [comment]
First trunk:
![image](https://user-images.githubusercontent.com/20173635/65042295-f09c2a00-d92e-11e9-8a71-e0f57349339b.png)
Second trunk:
![image](https://user-images.githubusercontent.com/20173635/65042306-f5f97480-d92e-11e9-847c-c5bc1f891cb8.png)
Rate group with both selected:
![image](https://user-images.githubusercontent.com/20173635/65042316-fa259200-d92e-11e9-8c96-33409157be85.png)

**Comment by MrSupaMike** [comment]
Hello @HorselessName ,

As I said, I only tried this on version 3.6 and it's working correctly. I assume that hasn't changed in v+4. 
One thing I suggest doing is to check on your freeswitch logs, in the linux console "fs_cli", you can easily check if it's working as the logs clearly state if translation is happening on each call.
Also, I noticed something that could be make a difference. The number translation format you use does not have "".  You could try adding them to see if it makes a difference. For example:

"1/3223#1"

If you need multiple translations, separate with commas like this:

"1/3223#1","9999/3223#"

![image](https://user-images.githubusercontent.com/35234441/65048357-951b6d80-d929-11e9-89d7-c9306c8ea941.png)


I have this currently working so I know it works on v3.6. I hope nothing changed on +v4

**Comment by HorselessName** [comment]
Ok. I tried selecting the two trunks and i also tried using the way you said, using "" and , to separate the prefixes... 

I found the problem. When the ASTPP 503 Service Unavailable, the Failover works...
But if the termination sends 603 Declined, the failover does'nt happen.
With 603 Declined the failover does'nt work...
When a termination returns to the ASTPP 603, the system does a hangup...

Is there a place on FreeSwitch or ASTPP where i can add more codes for the failover proccess?

**Comment by HorselessName** [comment]
Below a image of my monitoring (Ignore the REJECTED message, i just configured a test on my Asterisk to check what codes the ASTPP needs to receive in order to make the failover):

![image](https://user-images.githubusercontent.com/20173635/65277861-37da1480-db01-11e9-836b-22133def1065.png)
